clientv3/watch.go

```diff
@@ -46,7 +46,8 @@ type WatchResponse struct {
 	// CompactRevision is the minimum revision the watcher may receive.
 	CompactRevision int64
 
-	// Canceled is set to indicate the channel is about to close.
+	// If the watch failed and the stream was about to close, before the channel is closed,
+	// it returns a final response which has Canceled set to true and the error.
```

Comment:
the channel sends a final response that has Canceled set to true with a non-nil `Err()`